src/user-federation/UserFederationLdapSettings.tsx

```diff
@@ -15,16 +16,86 @@ import { LdapSettingsSynchronization } from "./ldap/LdapSettingsSynchronization"
 import { LdapSettingsGeneral } from "./ldap/LdapSettingsGeneral";
 import { LdapSettingsConnection } from "./ldap/LdapSettingsConnection";
 import { LdapSettingsSearching } from "./ldap/LdapSettingsSearching";
-import { ScrollForm } from "../components/scroll-form/ScrollForm";
 
-import { useHistory, useParams } from "react-router-dom";
 import { useRealm } from "../context/realm-context/RealmContext";
 import { convertToFormValues } from "../util";
-import { useAlerts } from "../components/alert/Alerts";
-import { useAdminClient } from "../context/auth/AdminClient";
 import ComponentRepresentation from "keycloak-admin/lib/defs/componentRepresentation";
 
-import { useForm } from "react-hook-form";
+import { Controller, useForm } from "react-hook-form";
+import { useConfirmDialog } from "../components/confirm-dialog/ConfirmDialog";
+import { useAdminClient } from "../context/auth/AdminClient";
+import { useAlerts } from "../components/alert/Alerts";
+import { useTranslation } from "react-i18next";
+import { ViewHeader } from "../components/view-header/ViewHeader";
+import { useHistory, useParams } from "react-router-dom";
+import { ScrollForm } from "../components/scroll-form/ScrollForm";
+
+type LdapSettingsHeaderProps = {
+  onChange: (...event: any[]) => void;
+  value: any;
```

Comment:
Avoid `any`

Reply:
I was getting errors unless using any. This is how it's done in clients and realms as well.

Reply:
This works but we could probably do better:
```suggestion
  onChange: (...event: unknown[]) => void;
  value: "true" | null;
```

Then you can also change line 183 to say:
`onChange={(value) => onChange(value)}`